### PR TITLE
Update examples for Babel, 0.14; remove 3rd party code

### PR DESCRIPTION
**Pull request by zpao**
There are a couple changes here. The most important are that this brings us up to compatibility with 0.14 (mostly just using `ReactDOM` where appropriate). The other thing is that we stop trying to use JSXTransformer, which we aren't even building so these are all broken right now. And lastly, I took this opportunity to stop bundling 3rd party code (es5-shim, sham, console, jquery, bootstrap) and just point at CDNs. This does mean that the examples will require an internet connection to actually run but I think that's ok - I don't care about the airplane case as much anymore.

I left the [server-rendering example](https://github.com/facebook/react/tree/master/examples/server-rendering) alone because I think I want to delete it. It's confusing to follow and run, and I think there are enough resources around on server rendering.

### CODE REVIEW

File: examples/basic-jsx-precompile/index.html [hunk 1]
```diff
@@ -11,23 +11,25 @@
       <p>
         If you can see this, React is not running. Try running:
       </p>
-      <pre>npm install -g react-tools
```
Comment (spicyj) [inline examples/basic-jsx-precompile/index.html]: What was wrong with this? Your way gives extra whitespace.

File: examples/jquery-bootstrap/js/app.js [hunk 2]
```diff
@@ -18,17 +18,17 @@ var BootstrapModal = React.createClass({
   // integrate Bootstrap or jQuery with the components lifecycle methods.
   componentDidMount: function() {
     // When the component is added, turn it into a modal
-    $(React.findDOMNode(this))
+    $(ReactDOM.findDOMNode(this))
```
Comment (spicyj) [inline examples/jquery-bootstrap/js/app.js]: Can you instead give a ref to the div and then not use findDOMNode?

**Comment by spicyj** [comment]
:+1:

File: examples/basic-jsx-precompile/index.html (hunk 1 above)
Comment (zpao) [inline examples/basic-jsx-precompile/index.html]: Nothing I just wanted prettier HTML. Comparing styling now though, looks like leaving as purely `pre` will be better.

**Pull request merged**